fix(tests): resolve resource leak in www/main_test.ts

fry69 opened the pull request:
The `buildVite` function returns a disposable object which was not being properly disposed of, leading to a resource leak warning in the test suite.

This change wraps the tests in `www/main_test.ts` in an async function and uses `await using` to ensure that the temporary directory created by `buildVite` is cleaned up after the tests complete.

google-labs-jules[bot] committed:
fix(tests): resolve resource leak in www/main_test.ts

The `buildVite` function returns a disposable object which was not being properly disposed of, leading to a resource leak warning in the test suite.

This change wraps the tests in `www/main_test.ts` in an async function and uses `await using` to ensure that the temporary directory created by `buildVite` is cleaned up after the tests complete.

fry69 committed:
chore: add dummy comment to trigger CI again

iuioiua commented on www/main_test.ts:
```diff
@@ -6,88 +6,93 @@ import { expect } from "@std/expect";
 import { buildVite } from "../packages/plugin-vite/tests/test_utils.ts";
 
-const result = await buildVite(import.meta.dirname!);
+async function runTests() {
```
Try [`Deno.test.beforeAll`](https://docs.deno.com/runtime/fundamentals/testing/#available-hooks) instead.

fry69 replied:
I do not know if using more Deno 2.5.0 isms is the way, people are already complaining that the Fresh 2 does not work with Deno LTS (2.2.14), see here -> https://github.com/denoland/deno/issues/30712

marvinhagemeister replied:
It's fine imo for our own test suite. User's don't typically run our test suite.

marvinhagemeister committed:
fix(vite): keep require calls in ESM file (#3433)

Detect if a file is ESM based on file extension or finding ESM-only
statements. When we know that it's an ESM file we should keep
`require()` calls as is, because some libraries use it as a conditional
check like `npm:motion`.

Fixes https://github.com/denoland/fresh/issues/3429

marvinhagemeister committed:
fix: bring back support for middleware Array (#3434)

Our validation didn't account for `_middleware` files exporting an Array
of middlewares.

Fixes https://github.com/denoland/fresh/issues/3400

emilkrebs committed:
feat: add CSP and header middleware (#3421)

Added a CSP and header middleware to simplify the HTTP Security Response
Header creation.

Fixes:
- https://github.com/denoland/fresh/issues/1705

Example:

```ts
// main.ts

export const app = new App<State>()
    .use(csp({
        reportTo: '/api/csp-reports',
    }))
    .use(headers({
        'Cross-Origin-Resource-Policy': 'same-site',
        'Cross-Origin-Embedder-Policy': 'require-corp',
        'Cross-Origin-Opener-Policy': 'same-origin',
        'Permissions-Policy':
            'geolocation=(), camera=(), microphone=(), payment=(), usb=(), bluetooth=(), magnetometer=(), gyroscope=(), accelerometer=()',
        'Referrer-Policy': 'strict-origin-when-cross-origin',
        'X-Content-Type-Options': 'nosniff',
        'X-Frame-Options': 'DENY',
        'Strict-Transport-Security': 'max-age=63072000; includeSubDomains; preload',
    }))
```

Endpoint example:
```ts
// routes/api/csp-handler.ts

import { Context } from 'fresh';
import { State } from '../../main.ts';

export const handler = {
    async POST(ctx: Context<State>) {
        try {
            const text = await ctx.req.text();
            const body = JSON.parse(text);

            const logEntry = {
                timestamp: new Date().toISOString(),
                report: body,
                userAgent: ctx.req.headers.get('user-agent') || 'unknown',
            };
            await Deno.writeFile('csp-reports.log', new TextEncoder().encode(JSON.stringify(logEntry) + '\n'), { append: true });

            return new Response(JSON.stringify({ status: "ok" }), { status: 200 });
        } catch (error) {
            console.error("Error handling CSP report:", error);
            return new Response(JSON.stringify({ status: "error", error: String(error) }), { status: 500 });
        }
    },
};

```

Implementing these headers improved the security report at [Security
Headers](securityheaders.com) from F to A and helps to prevent security
vulnerabilities like Cross-Site Scripting and Clickjacking.

**Before**
<img width="1200" height="308" alt="image"
src="https://github.com/user-attachments/assets/01adfff4-5bb4-4cc6-bd8f-5e74b471301b"
/>
**After**
<img width="1200" height="335" alt="image"
src="https://github.com/user-attachments/assets/c5a9b07e-babf-4797-b47d-4368de70cbc3"
/>

---------

Co-authored-by: Marvin Hagemeister <marvin@deno.com>

fry69 committed:
docs: fix minor issues in middleware docs (#3436)

fry69 committed:
test: convert to beforeAll()

fry69 committed:
tests: convert to two steps test

fry69 committed:
Merge branch 'main' into fix-test-resource-leak

fry69 committed:
tests: fix resource leaks

fry69 committed:
Merge branch 'main' into fix-test-resource-leak

fry69 committed:
tests: still trying to fix the resource leak error

marvinhagemeister commented on www/main_test.ts:
```diff
@@ -24,70 +41,71 @@ Deno.test("CORS should not set on GET /fresh-badge.svg", async () => {
-            await page.waitForSelector("#version");
+  sanitizeOps: false,
+  sanitizeResources: false,
+}, async () => {
```
Let's reduce indentation noise by moving the test function into the options object `fn: async () => {`.

fry69 replied:
Done.

fry69 committed:
tests: simplify test as requested

marvinhagemeister committed:
simplify

marvinhagemeister reviewed (approved):
LGTM
